### PR TITLE
chore: release v0.4.0

### PR DESCRIPTION
**Pull request by frol**
## 🤖 New release
* `near-api`: 0.3.0 -> 0.4.0 (⚠️ API breaking changes)

### ⚠️ `near-api` breaking changes

```
--- failure enum_missing: pub enum removed or renamed ---

Description:
A publicly-visible enum cannot be imported by its prior path. A `pub use` may have been removed, or the enum itself may have been renamed or removed entirely.
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#item-remove
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.38.0/src/lints/enum_missing.ron

Failed in:
  enum near_api::prelude::EpochReference, previously in file /tmp/.tmpGaArWl/near-api/src/types/reference.rs:43
  enum near_api::prelude::Reference, previously in file /tmp/.tmpGaArWl/near-api/src/types/reference.rs:10

--- failure enum_variant_added: enum variant added on exhaustive enum ---

Description:
A publicly-visible enum without #[non_exhaustive] has a new variant.
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#enum-variant-new
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.38.0/src/lints/enum_variant_added.ron

Failed in:
  variant ExecuteTransactionError:TransactionError in /tmp/.tmpxj58YP/near-api-rs/src/errors.rs:196

--- failure enum_variant_missing: pub enum variant removed or renamed ---

Description:
A publicly-visible enum has at least one variant that is no longer available under its prior name. It may have been renamed or removed entirely.
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#item-remove
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.38.0/src/lints/enum_variant_missing.ron

Failed in:
  variant ExecuteTransactionError::RetriesExhausted, previously in file /tmp/.tmpGaArWl/near-api/src/errors.rs:186

--- failure function_missing: pub fn removed or renamed ---

Description:
A publicly-visible function cannot be imported by its prior path. A `pub use` may have been removed, or the function itself may have been renamed or removed entirely.
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#item-remove
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.38.0/src/lints/function_missing.ron

Failed in:
  function near_api::signer::get_signed_delegate_action, previously in file /tmp/.tmpGaArWl/near-api/src/signer/mod.rs:272

--- failure inherent_method_missing: pub method removed or renamed ---

Description:
A publicly-visible method or associated fn is no longer available under its prior name. It may have been renamed or removed entirely.
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#item-remove
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.38.0/src/lints/inherent_method_missing.ron

Failed in:
  Signer::seed_phrase, previously in file /tmp/.tmpGaArWl/near-api/src/signer/mod.rs:167
  Signer::secret_key, previously in file /tmp/.tmpGaArWl/near-api/src/signer/mod.rs:178
  Signer::seed_phrase_with_hd_path, previously in file /tmp/.tmpGaArWl/near-api/src/signer/mod.rs:182
  Signer::access_keyfile, previously in file /tmp/.tmpGaArWl/near-api/src/signer/mod.rs:191
  Signer::ledger, previously in file /tmp/.tmpGaArWl/near-api/src/signer/mod.rs:196
  Signer::ledger_with_hd_path, previously in file /tmp/.tmpGaArWl/near-api/src/signer/mod.rs:201
  Signer::keystore, previously in file /tmp/.tmpGaArWl/near-api/src/signer/mod.rs:206
  Signer::keystore_search_for_keys, previously in file /tmp/.tmpGaArWl/near-api/src/signer/mod.rs:211

--- failure module_missing: pub module removed or renamed ---

Description:
A publicly-visible module cannot be imported by its prior path. A `pub use` may have been removed, or the module may have been renamed, removed, or made non-public.
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#item-remove
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.38.0/src/lints/module_missing.ron

Failed in:
  mod near_api::prelude, previously in file /tmp/.tmpGaArWl/near-api/src/lib.rs:17

--- failure pub_module_level_const_missing: pub module-level const is missing ---

Description:
A public const is missing, renamed, or changed from const to static.
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#item-remove
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.38.0/src/lints/pub_module_level_const_missing.ron

Failed in:
  W_NEAR_BALANCE in file /tmp/.tmpGaArWl/near-api/src/types/tokens.rs:8
  USDT_BALANCE in file /tmp/.tmpGaArWl/near-api/src/types/tokens.rs:6

--- failure struct_missing: pub struct removed or renamed ---

Description:
A publicly-visible struct cannot be imported by its prior path. A `pub use` may have been removed, or the struct itself may have been renamed or removed entirely.
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#item-remove
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.38.0/src/lints/struct_missing.ron

Failed in:
  struct near_api::prelude::Signer, previously in file /tmp/.tmpGaArWl/near-api/src/signer/mod.rs:91
  struct near_api::prelude::Account, previously in file /tmp/.tmpGaArWl/near-api/src/account/mod.rs:24
  struct near_api::prelude::Chain, previously in file /tmp/.tmpGaArWl/near-api/src/chain.rs:12
  struct near_api::prelude::StorageDeposit, previously in file /tmp/.tmpGaArWl/near-api/src/storage.rs:14
  struct near_api::prelude::Delegation, previously in file /tmp/.tmpGaArWl/near-api/src/stake.rs:30
  struct near_api::prelude::NetworkConfig, previously in file /tmp/.tmpGaArWl/near-api/src/config.rs:2
  struct near_api::prelude::FastNear, previously in file /tmp/.tmpGaArWl/near-api/src/fastnear.rs:63
  struct near_api::prelude::Data, previously in file /tmp/.tmpGaArWl/near-api/src/types/mod.rs:23
  struct near_api::prelude::Staking, previously in file /tmp/.tmpGaArWl/near-api/src/stake.rs:250
  struct near_api::prelude::FTBalance, previously in file /tmp/.tmpGaArWl/near-api/src/types/tokens.rs:11
  struct near_api::prelude::Contract, previously in file /tmp/.tmpGaArWl/near-api/src/contract.rs:27
  struct near_api::prelude::Transaction, previously in file /tmp/.tmpGaArWl/near-api/src/transactions.rs:71
  struct near_api::prelude::Tokens, previously in file /tmp/.tmpGaArWl/near-api/src/tokens.rs:38

--- failure trait_missing: pub trait removed or renamed ---

Description:
A publicly-visible trait cannot be imported by its prior path. A `pub use` may have been removed, or the trait itself may have been renamed or removed entirely.
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#item-remove
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.38.0/src/lints/trait_missing.ron

Failed in:
  trait near_api::prelude::SignerTrait, previously in file /tmp/.tmpGaArWl/near-api/src/signer/mod.rs:52
```

<details><summary><i><b>Changelog</b></i></summary><p>

<blockquote>

### Added

- added ability to specify backup rpc for connecting to the network (#28)
- don't retry on critical errors (query, tx) (#27)

### Other

- updates near-* dependencies to 0.28 release. Removed Cargo.lock (#33)
- [**breaking**] added documentation for root level and signer module (#32)
- added CODEOWNERS (#31)
- removed prelude and filtered entries.  (#29)
- replaced SecretBuilder with utility functions (#26)
- [**breaking**] replaced deploy method as a static method (#18)
</blockquote>

</p></details>

---
This PR was generated with [release-plz](https://github.com/release-plz/release-plz/).

### CODE REVIEW

**Comment by frol** [comment]
@race-of-sloths include my pr

**Comment by race-of-sloths** [comment]
@frol Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
You are true leader! Other sloths hot on your heels, keep moving!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/frol?type=bot&pr=near/near-api-rs/24&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/frol?type=bot&pr=near/near-api-rs/24&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/frol)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/frol)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Review by akorchyn — APPROVED** [review]

**Comment by race-of-sloths** [comment]
🥁  __Score it!__

@akorchyn, please score the PR with `@race-of-sloths score [1/2/3/5/8/13]`. The contributor deserves it.
If no scoring is provided within 24 hours, this PR will be scored as 1 🦥

**Pull request merged**
